New Dapr Components Metadata Format + Tutorial on Referencing Secrets in Dapr Components (#1234)
radius-project/docs: docs/content/guides/author-apps/dapr/how-to-dapr-secrets/index.md

---
type: docs
title: "How-To: Reference secrets in Dapr components"
linkTitle: "Reference secrets in components"
description: "Learn how to manage secrets in Dapr components"
weight: 300
categories: "How-To"
tags: ["Dapr"]
---

> This how-to guide will provide an overview of how to:

Reviewer: ```suggestion
This guide will provide an overview of how to:
```

- Securely manage secrets in Dapr components using [Dapr secret stores](https://docs.dapr.io/operations/components/setup-secret-store/)

## Prerequisites

- [rad CLI]({{< ref "installation#step-1-install-the-rad-cli" >}})
- [Bicep VSCode extension]({{< ref "installation#step-2-install-the-vs-code-extension" >}})
- [Radius environment]({{< ref "installation#step-3-initialize-radius" >}})
- [Radius local-dev Recipes]({{< ref howto-dev-recipes >}})
- [Dapr installed on your Kubernetes cluster](https://docs.dapr.io/operations/hosting/kubernetes/kubernetes-deploy/)

> ## Step 1: Start with a container and a Dapr sidecar

Reviewer: ```suggestion
## Step 1: Create a container and a Dapr sidecar
```

> Begin by creating a file named app.bicep, which defines a container with a Dapr state store. If you need a detailed explanation on how to do so, refer to the [Add a Dapr building block]({{< ref how-to-dapr-building-block >}}) tutorial.

Reviewer: ```suggestion
Begin by creating a file named `app.bicep`, which defines a container with a Dapr state store. If you need a detailed explanation on how to do so, refer to the [Add a Dapr building block]({{< ref how-to-dapr-building-block >}}) tutorial.
```
